### ISSUE TITLE
for speeding up

### ISSUE THREAD

**Issue by smichel** [smichel/py_rainprog#4]
https://numba.pydata.org/

**Comment by smichel** [comment]
from sklearn.feature_extraction import image
np.sum((image.extract_patches_2d(a, b.shape)-b)**2, axis=(1,2)).reshape(np.array(a.shape)-b.shape+1)
für leastsquarecorr

**Comment by smichel** [comment]
@jit(nopython=True)
def interp2d(nested_data, x, y):  # nested_data in 2d
    vals = np.empty_like(x)
    for idx, xx in np.ndenumerate(x):
        xi = int(xx)
        xmod = xx % 1
        yy = y[idx]
        yi = int(yy)
        ymod = yy % 1

        vals[idx] = nested_data[xi, yi] * ((1 - xmod) * (1 - ymod)) + \
             nested_data[xi, yi+1] * (1- xmod) * (ymod) + \
             nested_data[xi+1, yi+1] * ((xmod) * ymod) + \
             nested_data[xi+1, yi] * ((xmod) * (1 - ymod))
    return vals
